### PR TITLE
RSDK-3124 - Board server side

### PR DESCRIPTION
**Pull request by clintpurser**
This PR adds the server side for Board.

Testing will be added in a follow up and is tracked here https://viam.atlassian.net/browse/RSDK-3283

### CODE REVIEW

**Comment by clintpurser** [comment]
CI will fail till https://github.com/viamrobotics/viam-flutter-sdk/pull/27 is in.

File: lib/src/components/board/client.dart [hunk 1]
```diff
@@ -15,6 +16,12 @@ class BoardClient extends Board {
 
   BoardClient(this.name, this._channel) : _client = BoardServiceClient(_channel);
 
+  @override
+  Future<Map<String, dynamic>> doCommand(Map<String, dynamic>? command) async {
+    final response = await _client.doCommand(common.DoCommandRequest(name: name, command: command?.toStruct()));
+    return response.result.toMap();
+  }
+
```
Comment (clintpurser) [inline lib/src/components/board/client.dart:24]: driveby: board didn't have doCommand implemented

File: lib/src/components/board/board.dart [hunk 2]
```diff
@@ -16,6 +18,13 @@ class BoardStatus {
     pbBoardStatus.digitalInterrupts.forEach((key, value) => boardStatus.digitalInterrupts[key] = (value.value.toInt()));
     return boardStatus;
   }
+
+  common.BoardStatus toPbBoardStatus() {
```
Comment (njooma) [inline lib/src/components/board/board.dart:22]: I think this would be better as a ` get` attribute called `proto`

```suggestion
  common.BoardStatus get proto {
```

File: lib/src/components/board/board.dart [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import 'package:fixnum/src/int64.dart';
+import 'package:viam_sdk/src/proto/common.dart';
 import 'package:viam_sdk/src/resource/base.dart';
```
Comment (njooma) [inline lib/src/components/board/board.dart:3]: These should be relative imports

File: lib/src/components/board/client.dart [hunk 4]
```diff
@@ -2,6 +2,7 @@ import 'package:fixnum/src/int64.dart';
 import 'package:grpc/grpc_connection_interface.dart';
 import 'package:viam_sdk/src/gen/google/protobuf/duration.pb.dart' as grpc_duration;
```
Comment (njooma) [inline lib/src/components/board/client.dart:3]: Relative import

**Review by njooma — CHANGES_REQUESTED** [review]
Small changes

File: lib/src/components/board/board.dart (hunk 2 above)
Comment (clintpurser) [reply lib/src/components/board/board.dart:22]: i like that, good call.

File: lib/src/components/board/board.dart (hunk 3 above)
Comment (clintpurser) [reply lib/src/components/board/board.dart:3]: ah yeah, thanks. 👀

File: lib/src/components/base/client.dart [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import 'package:fixnum/src/int64.dart';
+import 'package:fixnum/fixnum.dart';
 import 'package:grpc/grpc_connection_interface.dart';
 import 'package:viam_sdk/src/gen/common/v1/common.pb.dart';
 import 'package:viam_sdk/src/gen/component/base/v1/base.pbgrpc.dart';
```
Comment (njooma) [inline lib/src/components/base/client.dart:4]: Local imports :)

**Review by njooma — APPROVED** [review]
Nit on imports

File: lib/src/components/base/client.dart (hunk 5 above)
Comment (clintpurser) [reply lib/src/components/base/client.dart:4]: noooo i thought i got these 😭

Comment (clintpurser) [reply lib/src/components/base/client.dart:4]: oh this is in base 👀

Comment (njooma) [reply lib/src/components/base/client.dart:4]: Oh that's on me then 😬
